app.js

```diff
@@ -145,14 +137,24 @@ var controller = (function(budgetCtrl, uiCtrl) {
         this.description = description,
         this.number = number
     }
+    var buttonAncestor;
 
     document.querySelector('.add__btn').addEventListener("click", clickedAddButton);
-    //document.querySelector('.item__delete--btn').addEventListener('click', uiCtrl.deleteItem);
+    document.querySelector('.income__list').addEventListener("click", clickedDeleteButton);
+    document.querySelector('.expenses__list').addEventListener("click", clickedDeleteButton);
 
     function clickedAddButton () {
         var newItem = new Transaction(document.querySelector('.add__type').value, document.querySelector('.add__description').value, document.querySelector('.add__value').value);
         budgetCtrl.updateNumbers(newItem);
         uiCtrl.updateUi(newItem, budgetCtrl.totalIncome, budgetCtrl.totalExpenses, budgetCtrl.availableBudget,
             budgetCtrl.itemPercentOfTotalIncome, budgetCtrl.expensePercentOfTotalIncome, budgetCtrl.newPercentages);
     }
+
+    function clickedDeleteButton () {
+        if (event.target.nodeName == 'I') {
+            buttonAncestor = event.target.parentElement.parentElement.parentElement.parentElement.id; // Get the id of the item containing the clicked button.
+            console.log(buttonAncestor);
+            // Pass buttonAncestor to the budget controller for calculations.
+        }
+    }

```

Comment:
Lessons learned:
• Use parentElement or parentNode to get the parent of an element. Since they return element objects, I can use id, classname, etc to get additional details about the parent. I can chain parents together to get elements that are higher in the DOM.
• To add a click event to the delete buttons, which don't exist in the DOM until the user adds an item, I added an event listener to the divs that already existed and would contain the buttons.